typescript-react/src/components/Table/Table.tsx

```diff
@@ -218,14 +219,22 @@ export class Table<T> extends React.Component<ITableProps<T>, ITableState<T>> {
   }
 
   private renderPagination = (props: any) => {
-    const { data, BeforePaginationComponent, totalCount } = this.props;
+    const { data, BeforePaginationComponent, pageSizeOptions, totalCount } = this.props;
     return BeforePaginationComponent ? (
       <React.Fragment>
         { data && data.length > 0 ? <BeforePaginationComponent /> : null}
-        <Pagination {...props} totalCount={totalCount} />
+        <Pagination
+          {...props}
+          pageSizeOptions={pageSizeOptions || props.pageSizeOptions}
```

Comment:
how is pageSizeOptions || props.pageSizeOptions
different then just pageSizeOptions (as you have taken it from this.props)

Reply:
The `renderPagination` method takes `props` as an argument the props are provided by `ReactTable`. The `pageSizeOptions` that we use here is from `this.props` which are provided by passing props to our `Table` component.